avm/res/container-service/managed-cluster/main.bicep

```diff
@@ -0,0 +1,1012 @@
+  podSubnetId: string?
+
+  @description('Optional. The proximity placement group resource ID of the agent pool.')
+  proximityPlacementGroupResourceId: string?
+
+  @description('Optional. The scale down mode of the agent pool.')
+  scaleDownMode: ('Delete' | 'Pause' | 'Requeue' | 'DeleteRequeue')?
+
+  @description('Optional. The scale set eviction policy of the agent pool.')
+  scaleSetEvictionPolicy: ('Delete' | 'Deallocate')?
+
+  @description('Optional. The scale set priority of the agent pool.')
+  scaleSetPriority: ('Low' | 'Regular' | 'Spot')?
+
+  @description('Optional. The spot max price of the agent pool.')
+  spotMaxPrice: int?
+
+  @description('Optional. The tags of the agent pool.')
+  tags: object?
+
+  @description('Optional. The type of the agent pool.')
+  type: ('AvailabilitySet' | 'VirtualMachineScaleSets')?
+
+  @description('Optional. The maximum number of nodes that can be created during an upgrade.')
+  maxSurge: string?
+
+  @description('Optional. The VM size of the agent pool.')
+  vmSize: string?
+
+  @description('Optional. The VNet subnet ID of the agent pool.')
+  vnetSubnetID: string?
+
+  @description('Optional. The workload runtime of the agent pool.')
+  workloadRuntime: string?
+
+  @description('Optional. The enable default telemetry of the agent pool.')
+  enableDefaultTelemetry: bool?
+
+  @description('')
+  storageProfile: string?
+
+}[]?
+
+type managedIdentitiesType = {
+  @description('Optional. Enables system assigned managed identity on the resource.')
+  systemAssigned: bool?
+
+  @description('Optional. The resource ID(s) to assign to the resource.')
+  userAssignedResourcesIds: string[]?
+}?
+
+type lockType = {
+  @description('Optional. Specify the name of lock.')
+  name: string?
+
+  @description('Optional. Specify the type of lock.')
+  kind: ('CanNotDelete' | 'ReadOnly' | 'None')?
+}?
+
+type roleAssignmentType = {
+  @description('Required. The name of the role to assign. If it cannot be found you can specify the role definition ID instead.')
+  roleDefinitionIdOrName: string
+
+  @description('Required. The principal ID of the principal (user/group/identity) to assign the role to.')
+  principalId: string
+
+  @description('Optional. The principal type of the assigned principal ID.')
+  principalType: ('ServicePrincipal' | 'Group' | 'User' | 'ForeignGroup' | 'Device')?
+
+  @description('Optional. The description of the role assignment.')
+  description: string?
+
+  @description('Optional. The conditions on the role assignment. This limits the resources it can be assigned to. e.g.: @Resource[Microsoft.Storage/storageAccounts/blobServices/containers:ContainerName] StringEqualsIgnoreCase "foo_storage_container"')
+  condition: string?
+
+  @description('Optional. Version of the condition.')
+  conditionVersion: '2.0'?
+
+  @description('Optional. The Resource Id of the delegated managed identity resource.')
+  delegatedManagedIdentityResourceId: string?
+}[]?
```

Comment:
```suggestion
type roleAssignmentType = {
  @description('Required. The role to assign. You can provide either the display name of the role definition, the role definition GUID, or its fully qualified ID in the following format: \'/providers/Microsoft.Authorization/roleDefinitions/c2f4ef07-c644-48eb-af81-4b1b4947fb11\'.')
  roleDefinitionIdOrName: string

  @description('Required. The principal ID of the principal (user/group/identity) to assign the role to.')
  principalId: string

  @description('Optional. The principal type of the assigned principal ID.')
  principalType: ('ServicePrincipal' | 'Group' | 'User' | 'ForeignGroup' | 'Device')?

  @description('Optional. The description of the role assignment.')
  description: string?

  @description('Optional. The conditions on the role assignment. This limits the resources it can be assigned to. e.g.: @Resource[Microsoft.Storage/storageAccounts/blobServices/containers:ContainerName] StringEqualsIgnoreCase "foo_storage_container".')
  condition: string?

  @description('Optional. Version of the condition.')
  conditionVersion: '2.0'?

  @description('Optional. The Resource Id of the delegated managed identity resource.')
  delegatedManagedIdentityResourceId: string?
}[]?
```